### PR TITLE
Fix `IllegalStateException` error when the external client was already started

### PR DESCRIPTION
**Pull request by snaiper80**
Motivation
---
If a user wants to use an external `Curator` client in`ZooKeeperUpdatingListener` and the `Curator` instance was already started. The user can see the following exception.

```
java.lang.IllegalStateException: Cannot be started more than once
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.start(CuratorFrameworkImpl.java:311)
	at com.linecorp.armeria.server.zookeeper.ZooKeeperUpdatingListener.serverStarted(ZooKeeperUpdatingListener.java:123)
	at com.linecorp.armeria.server.Server$ServerStartStopSupport.notifyStarted(Server.java:658)
	at com.linecorp.armeria.server.Server$ServerStartStopSupport.notifyStarted(Server.java:400)
	at com.linecorp.armeria.common.util.StartStopSupport.notifyListeners(StartStopSupport.java:374)
	at com.linecorp.armeria.common.util.StartStopSupport.enter(StartStopSupport.java:362)
	at com.linecorp.armeria.common.util.StartStopSupport.lambda$start0$6(StartStopSupport.java:221)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478)
	at io.netty.util.concurrent.GlobalEventExecutor$TaskRunner.run(GlobalEventExecutor.java:243)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)
``` 

I want to fix this issue.

Modifications
---
* call start() method when the Curator client is not `STARTED` state.

### CODE REVIEW

File: zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java [hunk 1]
```diff
@@ -119,7 +120,9 @@ public static ZooKeeperUpdatingListenerBuilder builder(
     @Override
     public void serverStarted(Server server) throws Exception {
         final ZooKeeperRegistrationSpec registrationSpec = fillAndCreateNewRegistrationSpec(spec, server);
-        client.start();
+        if (client.getState() != CuratorFrameworkState.STARTED) {
```
Comment (ikhoon) [inline zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java:123]: Oop... I didn't think of this case.

**Review by ikhoon — APPROVED** [review]
Thanks! @snaiper80

File: zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java (hunk 1 above)
Comment (ikhoon) [reply zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java:123]: Could you add this checking to the constructor of `ZooKeeperEndpointGroup`?

Comment (snaiper80) [reply zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java:123]: I also don't know this case before using an external instance is set. T.T

File: zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package com.linecorp.armeria.server.zookeeper;
```
Comment (minwoox) [inline zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java:1]: We need the license. 🤣

File: zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                                    .http(endpoint.port())
+                                    .service("/", (ctx, req) -> HttpResponse.of(200))
+                                    .build();
+        server.addListener(listener);
+        startServerWithRetries(server);
+        await().untilAsserted(() -> zkInstance.assertExists(Z_NODE + "/foo/bar"));
+        server.stop().join();
+        client.close();
+    }
+
```
Comment (minwoox) [inline zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java:57]: I think this also violates the check style so we should remove this extra empty line. 😄

File: zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java (hunk 1 above)
Comment (snaiper80) [reply zookeeper3/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java:123]: `ZooKeeperEndpointGroup`? Got it. will check it.

**Review by minwoox — COMMENTED** [review]
Could you also apply the same approach for `ZooKeeperEndpointGroup`, please?
😄

**Comment by minwoox** [comment]
> Could you also apply the same approach for ZooKeeperEndpointGroup, please?
😄

I missed @ikhoon comment. 😅

File: zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java (hunk 2 above)
Comment (snaiper80) [reply zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java:16]: Ah. Missed. will add it.

File: zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java (hunk 3 above)
Comment (snaiper80) [reply zookeeper3/src/test/java/com/linecorp/armeria/server/zookeeper/CuratorServiceExternalClientUsageTest.java]: Got it.

**Review by minwoox — APPROVED** [review]
Thanks a lot, @snaiper80!

**Comment by snaiper80** [comment]
Thank you! @minwoox @ikhoon

**Review by trustin — APPROVED** [review]
Thanks, @snaiper80 ! :bow:

**Pull request merged**